Update GameManager done

DullySkull opened the pull request:
Pause work, Game Over works
Singelton exist already and gives error need help tomarrow new scenes created for turtorial and main menu

Liighht committed:
movement adjustments

i adjusted movement parameters for the player, i attached him to the train for now i don't have issues with it and it feels better, dive force feels a bit hard coded. I changed the dash button to spacebar, more intuitive.

turnercore committed:
Merge branch 'main' into Design

tibetyuksel committed:
Tibet_Map_Created

Rinurfriend committed:
Added new Geo_robot

- Pivot fixes
- Added hands
-

Stengolem committed:
The Title

Added the 3D title in Finished Prefabs for the main menu

Stengolem committed:
Merge branch 'Artists' of https://github.com/ltsLumina/FG-GP-Team1 into Artists

turnercore committed:
Designer Sandbox and prefab variables for speed update

jakhenn committed:
Added coral and fuel resource assets

Added models textures materials and prefabs

turnercore committed:
Merge remote-tracking branch 'origin/Artists' into turner

turnercore committed:
Merge branch 'Design' into turner

turnercore committed:
Null check for spawner

turnercore committed:
Spanwer Fixes

turnercore committed:
Moved some stuff around

turnercore committed:
Player Animations

turnercore committed:
Slight Player Anim Change

Just moved around some layers

turnercore committed:
Merge branch 'main' into turner

turnercore committed:
commented out the currently unimplimented animations

DullySkull committed:
Update GameManager done

Pause work, Game Over works
Singelton exist already and gives error need help tomarrow new scenes created for turtorial and main menu

DullySkull committed:
Update GameManager done

Pause work, Game Over works
Singelton exist already and gives error need help tomarrow new scenes created for turtorial and main menu

ltsLumina commented on Assets/_Project/Runtime/_Scripts/GameManager/GameManager.cs:
```diff
@@ -0,0 +1,135 @@
+        switch (state)
+        {
+            case GameState.Play:
+                if (Input.GetKeyDown(KeyCode.Escape))
```
I'll convert all of this to the new InputSystem eventually

DullySkull replied:
👍

ltsLumina commented on Assets/_Project/Runtime/_Scripts/GameManager/TurtorialManager.cs:
```diff
@@ -0,0 +1,132 @@
+    {
+        switch (state)
+        {
+            case GameState.Play:
+                if (Input.GetKeyDown(KeyCode.Escape))
+                {
+                    GameStateChanger(GameState.Pause);
+                    pausePanel.SetActive(true);
+                }
+                break;
+            case GameState.Pause:
+                if (Input.GetKeyDown(KeyCode.Escape))
+                {
+                    GameStateChanger(GameState.Play);
+                    pausePanel.SetActive(false);
+                }
+                break;
+            case GameState.Movement:
+                // Player try movement
+                break;
+            case GameState.Pickup:
+                // Player try pickup
+                break;
+            case GameState.Fuel:
+                // Player try fuel
+                break;
+            case GameState.Repair:
+                // Player try repair
+                break;
+            case GameState.Recharge:
+                // Player try recharge
+                break;
+            case GameState.TurtorialDone:
+                // Change scene
+                break;
+        }
+        // Add more cases if needed
+
+        GameStateChanger(state);
+    }
```
Love this. Good job :)

DullySkull replied:
😄

ltsLumina commented on Assets/_Project/Runtime/_Scripts/UI Scripts/ScoreManager.cs:
```diff
@@ -0,0 +1,62 @@
+    [SerializeField] TextMeshProUGUI currentLifeformsScanned;
+
+    [Header("Current Scores")]
+    public int depthScore;
+    public int algaeCollected;
+    public int holesRepaired;
+    public int jellyfishCollected;
+    public int lifeformsScanned;
```
Don't make public variables. Use properties instead.
Example:

```c#
[SerializeField] int depthScore;

public int DepthScore => depthScore; // get only property

// or:

public int DepthScore
{
    get => depthScore;
    set => depthScore = value;
}
```

DullySkull replied:
Fixed it! Did the same thing with high score

ltsLumina reviewed (changes requested):
Looks good!
If you could just change the public variables to be separate properties that would be great :)
I'll take a look tomorrow again before we make the Alpha build!

ltsLumina committed:
Merge pull request #16 from ltsLumina/Artists

Artists

ltsLumina committed:
Merge branch 'main' into turner

Commit:
Update Kelp Model

ltsLumina committed:
Merge pull request #17 from ltsLumina/turner

Animations into Main

Commit:
Added Animations, Updated Kelp, Seaweed, Added IDestructible, Optimized & Updated WallGen.cs

turnercore committed:
Renamed player animation component

turnercore committed:
Merge branch 'main' into turner

DullySkull committed:
Properties update from public variables

DullySkull committed:
Properties update from public variables

DullySkull committed:
Sorry forgot to update the highscoremanager after the changes

Removed the other gamemanager and put the konamicodechecker script on the other gamemanager so it dosen't error also fixed so it should connect with other script

DullySkull committed:
Sorry forgot to update the highscoremanager after the changes

Removed the other gamemanager and put the konamicodechecker script on the other gamemanager so it dosen't error also fixed so it should connect with other script

ltsLumina committed:
Merge pull request #19 from ltsLumina/turner

Turner

ltsLumina committed:
Walls with Mesh Colliders

ltsLumina committed:
Train lighting implemented. (No battery yet)

Nebrok committed:
Tweaking of Noise gen, and removal of unused

Nebrok committed:
Merge pull request #22 from ltsLumina/WallFix

Tweaking of Noise gen, and removal of unused

ltsLumina committed:
Battery WIP

DullySkull committed:
Update

Hopefully everything works ;-;

DullySkull committed:
Update

Hopefully everything works ;-;

ltsLumina committed:
nothing in particular

ltsLumina committed:
faeadw

ltsLumina committed:
Revert "faeadw"

This reverts commit eee56bd554bf04bbd5c6acb430221a4d5113fe91, reversing
changes made to a6f9c6635bedbe243ea9ee83dac05f972b8a5e87.

ltsLumina committed:
fixing

ltsLumina committed:
build profile

ltsLumina committed:
Final Main Menu UI Stuff
